backend/capellacollab/sessions/routes.py

```diff
@@ -364,6 +354,47 @@ def request_persistent_session(
     return response
 
 
+def determine_pure_variants_configuration(
+    db: Session, user: DatabaseUser, tool: Tool, warnings: list[Message]
```

Comment:
It would be nicer, more functional, if `warnings` is returned from the function.

Idealy all parameters are input. Try to avoid in/out parameters

Reply:
Good point. I've changed it :)